lib/readline.js

```diff
@@ -193,8 +211,27 @@ Interface.prototype._onLine = function(line) {
   } else {
     this.emit('line', line);
   }
+  if(this.historyPath) this._saveLine(line);
 };
 
+
+Interface.prototype._saveLine = function(line) {
+  try {
+    fs.appendFileSync(this.historyPath, line + '\n');
+  } catch (e) {}
+};
+
+
+Interface.prototype._saveHistory = function() {
+  var history = this.history.slice()
+    .reverse()
+    .join('\n');
+  try {
+    fs.writeFileSync(this.historyPath, history);
```

Comment:
We should not use the `*Sync` fs methods. Users might add a repl to their live server, and this would unduly affect the performance of those servers.